src/Compiler/Facilities/BuildGraph.fs

```diff
@@ -193,7 +193,7 @@ type NodeCode private () =
         }
 
     static member Parallel(computations: NodeCode<'T> seq) =
-        computations |> Seq.map (fun (Node x) -> x) |> Async.Parallel |> Node
+        computations |> Seq.map (fun (Node x) -> x) |> Async.Parallel |> wrapThreadStaticInfo |> Node
```

Comment:
@majocha the test passes for me with this change. 

However it doesn't mean that the `computations` will have the same thread statics as the parent computation. Which we could also be relying on somewhere.

Reply:
Correct, I'll update the test in a moment.

Reply:
That could be probably fixed like this:

```suggestion
        let diagnosticsLogger = DiagnosticsThreadStatics.DiagnosticsLogger
        let phase = DiagnosticsThreadStatics.BuildPhase

        computations
        |> Seq.map (fun (Node x) ->
            async {
                DiagnosticsThreadStatics.DiagnosticsLogger <- diagnosticsLogger
                DiagnosticsThreadStatics.BuildPhase <- phase
                return! x
            })
        |> Async.Parallel
        |> wrapThreadStaticInfo
        |> Node
```

Reply:
But actually looking at the places where it's currently used this is probably not relied on. (Also if it was we would probably have noticed it).